src/System.IO.FileSystem/tests/Directory/GetFiles_str_Str.cs

```diff
@@ -11,454 +12,274 @@
-            iCountTestcases++;
-            if (Array.IndexOf(names, "TestFile3") < 0)
-            {
-                iCountErrors++;
-                printerr("Error_879by! Incorrect name==" + strArr[2].ToString());
-            }
+    public virtual String[] GetFiles(String path, String searchPattern)
+    {
+        return Directory.GetFiles(path, searchPattern);
+    }
 
+    #endregion
 
-            // Search string is '*'
-            strLoc = "Loc_4y7gb";
+    #region UniversalTests
 
-            strArr = Directory.GetFiles(Path.Combine(".", dirName), "*");
-            iCountTestcases++;
-            if (strArr.Length != 5)
-            {
-                iCountErrors++;
-                printerr("Error_t5792! Incorrect number of files==" + strArr.Length);
-            }
-            names = new String[strArr.Length];
-            i = 0;
-            foreach (String f in strArr)
-                names[i++] = Path.GetFileName(f);
+    [Fact]
+    public void NullSearchPattern()
+    {
+        Assert.Throws<ArgumentNullException>(() => GetFiles(".", null));
+    }
 
-            if (!Interop.IsWindows) // test is expecting sorted order as provided by Windows
-            {
-                Array.Sort(names);
-            }
+    [Fact]
+    public void EmptySearchPattern()
+    {
+        Assert.Empty(GetFiles(".", String.Empty));
+    }
 
-            iCountTestcases++;
-            if (Array.IndexOf(names, "Test1File1") < 0)
-            {
-                iCountErrors++;
-                printerr("Error_4898v! Incorrect name==" + strArr[0].ToString());
-            }
-            iCountTestcases++;
-            if (Array.IndexOf(names, "Test1File2") < 0)
-            {
-                iCountErrors++;
-                printerr("Error_4598c! Incorrect name==" + strArr[1].ToString());
-            }
-            iCountTestcases++;
-            if (Array.IndexOf(names, "TestFile1") < 0)
-            {
-                iCountErrors++;
-                printerr("Error_209d8! Incorrect name==" + strArr[2].ToString());
-            }
-            iCountTestcases++;
-            if (Array.IndexOf(names, "TestFile2") < 0)
-            {
-                iCountErrors++;
-                printerr("Error_10vtu! Incorrect name==" + strArr[3].ToString());
-            }
-            iCountTestcases++;
-            if (Array.IndexOf(names, "TestFile3") < 0)
-            {
-                iCountErrors++;
-                printerr("Error_190vh! Incorrect name==" + strArr[4].ToString());
-            }
+    [Fact]
+    public void WhitespaceSearchPattern()
+    {
+        Assert.Empty(GetFiles(TestDirectory, "           "));
+        Assert.Empty(GetFiles(TestDirectory, "\n"));
+    }
 
-            strLoc = "Loc_4yg7b";
+    [Fact]
+    public void InvalidSearchPattern()
+    {
+        // '<' and '>' are valid Unix filename chars
+        if (Interop.IsWindows)
+        {
+            Assert.Throws<ArgumentException>(() => GetFiles(".", "<>"));
```

Comment:
> The distinction of allowing newlines by themselves but not in a name seems a bit inconsistent, unless there's a use case I'm missing.

I don't think it's that newlines alone are allowed and disallowed otherwise, but that various whitespace charactes (including '\n') are trimmed from the end of the search string (I don't know what the justification is for such trimming, but it's been the behavior for a long time).  So  "\n" and "something\n" are both valid, but "some\nthing" is not: the string is trimmed and then checked for invalid characters.